### ISSUE TITLE
Possible web services

### ISSUE THREAD

**Issue by gdude2002** [UltrosBot/Ultros-site#6]
I've been thinking about this for a while, and I feel like it might be nice to provide some extra centralized services to Ultros users.

We've been talking about the packages repo stuff for some time - this is separate from all of that for now.

* Accounts system that allows you to associate bots with your account
* Easy access to exceptions without needing the UUID
* Simple monitoring and control options
* OAuth management for stuff like Slack (this is the only way I can think of that's both easy + secure)
* Some kind of API

I'm wondering what other stuff we could add to this. Any ideas?

**Comment by rakiru** [comment]
It'd certainly be handy to get easy access to exceptions. I have DerpyBot and Ultros' pages bookmarked for easy access, but that's obviously not ideal.

I don't know enough about OAuth to be much use, but I feel we should at least make it easy to do without relying on ultros.io (web plugin for callback or w/e).

Personally, I'm against any form of 3rd party control of my server, so I'm not sure what kinds of things people might want. Also, the whole idea behind Ultros' design is that you only need one bot to do everything, so anything relating to multiple bots is outside of my use case anyway (and potentially out of scope).

**Comment by gdude2002** [comment]
> Personally, I'm against any form of 3rd party control of my server

Yeah, I 100% understand that - All control stuff will be entirely optional, incl. being able to have the other services operational without the control stuff. There'll be a flag bot-side that will make the bot reject any control requests, and since we're OSS, people can verify that themselves.

> ...anything relating to multiple bots is outside of my use case anyway (and potentially out of scope)

Yeah, agreed, but it doesn't make sense to me to *not* build in multi-bot while it's being written. Unnecessary limitations and all that.

**Comment by rakiru** [comment]
Of course it'll be optional. I just mean that I can't suggest features that I don't want.

Yes, I agree that you should definitely consider it while building it. My point is that I don't see much point adding lots of features that are *only* useful for multiple bots.

**Comment by gdude2002** [comment]
Ah, I see your point, alright. I'll bear that in mind.
